pkg/alertmanager/amcfg.go

```diff
@@ -869,3 +864,78 @@ func (cg *configGenerator) convertTLSConfig(ctx context.Context, in *monitoringv
 
 	return out
 }
+
+// sanitize the config against a specific AlertManager version
+// strips fields from config if unsupported
+func (c *alertmanagerConfig) sanitize(amVersion semver.Version, logger log.Logger) {
+	if c == nil {
+		return
+	}
+
+	if c.Global != nil {
+		c.Global.sanitize(amVersion, logger)
+	}
+
+	// We need to sanitize the config for slack receivers
+	// As of v0.22.0 AlertManager config supports passing URL via file name
+	fileURLAllowed := amVersion.GTE(semver.MustParse("0.22.0"))
+	for _, receiver := range c.Receivers {
+		for _, slackConfig := range receiver.SlackConfigs {
+			if slackConfig.APIURLFile == "" {
+				continue
+			}
+
+			if slackConfig.APIURL != "" {
+				msg := "'api_url' and 'api_url_file' are mutually exclusive for slack receiver config - 'api_url' has taken precedence"
+				level.Warn(logger).Log("msg", msg, "receiver", receiver.Name)
+				slackConfig.APIURLFile = ""
+			}
+
+			if !fileURLAllowed {
+				msg := "'api_url_file' supported in AlertManager >= 0.22.0 only - dropping field from provided config"
+				level.Warn(logger).Log("msg", msg, "current_version", amVersion.String(), "receiver", receiver.Name)
+				slackConfig.APIURLFile = ""
+			}
+		}
+	}
+}
+
+// sanitize globalConfig - gc must not be nil
+func (gc *globalConfig) sanitize(amVersion semver.Version, logger log.Logger) {
+	if gc.HTTPConfig != nil {
+		gc.HTTPConfig.sanitize(amVersion, logger)
```

Comment:
we need to sanitize all `*httpClientConfig` fields (e.g. for all receivers).

Reply:
I missed that they all had that conf :/